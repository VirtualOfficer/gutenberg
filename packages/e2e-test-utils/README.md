# E2E Test Utils

End-To-End (E2E) test utils for WordPress.

## Installation

Install the module

```bash
npm install @wordpress/e2e-test-utils --save-dev
```

## API

<!-- START TOKEN(Autogenerated API docs) -->

<a href="#activateplugin">#</a> **activatePlugin** [\<>](src/index.js#L1-L1)

Activates an installed plugin.

_Parameters_

-   `string`: Plugin slug.

<a href="#areprepublishchecksenabled">#</a> **arePrePublishChecksEnabled** [\<>](src/index.js#L2-L2)

Verifies if publish checks are enabled.

_Returns_

-   `boolean`: Boolean which represents the state of prepublish checks.

<a href="#clearlocalstorage">#</a> **clearLocalStorage** [\<>](src/index.js#L3-L3)

Clears the local storage.

<a href="#clickblockappender">#</a> **clickBlockAppender** [\<>](src/index.js#L4-L4)

Clicks the default block appender.

<a href="#clickblocktoolbarbutton">#</a> **clickBlockToolbarButton** [\<>](src/index.js#L5-L5)

Clicks a block toolbar button.

_Parameters_

-   `string`: The aria label of the button to click.

<a href="#clickbutton">#</a> **clickButton** [\<>](src/index.js#L6-L6)

Clicks a button based on the text on the button.

_Parameters_

-   `string`: The text that appears on the button to click.

<a href="#clickonclosemodalbutton">#</a> **clickOnCloseModalButton** [\<>](src/index.js#L7-L7)

Click on the close button of an open modal.

_Parameters_

-   `?string`: Class name for the modal to close

<a href="#clickonmoremenuitem">#</a> **clickOnMoreMenuItem** [\<>](src/index.js#L8-L8)

Clicks on More Menu item, searches for the button with the text provided and clicks it.

_Parameters_

-   `string`: The label to search the button for.

<a href="#createembeddingmatcher">#</a> **createEmbeddingMatcher** [\<>](src/index.js#L52-L52)

Creates a function to determine if a request is embedding a certain URL.

_Parameters_

-   `string`: The URL to check against a request.

_Returns_

-   `function`: Function that determines if a request is for the embed API, embedding a specific URL.

<a href="#createjsonresponse">#</a> **createJSONResponse** [\<>](src/index.js#L52-L52)

Respond to a request with a JSON response.

_Parameters_

-   `string`: The mock object to wrap in a JSON response.

_Returns_

-   `Promise`: Promise that responds to a request with the mock JSON response.

<a href="#createnewpost">#</a> **createNewPost** [\<>](src/index.js#L9-L9)

Creates new post.

_Parameters_

-   `Object`: Object to create new post, along with tips enabling option.

<a href="#createurl">#</a> **createURL** [\<>](src/index.js#L10-L10)

Creates new URL by parsing base URL, WPPath and query string.

_Parameters_

-   `string`: String to be serialized as pathname.
-   `?string`: String to be serialized as query portion of URL.

_Returns_

-   `string`: String which represents full URL.

<a href="#createurlmatcher">#</a> **createURLMatcher** [\<>](src/index.js#L52-L52)

Creates a function to determine if a request is calling a URL with the substring present.

_Parameters_

-   `string`: The substring to check for.

_Returns_

-   `function`: Function that determines if a request's URL contains substring.

<a href="#deactivateplugin">#</a> **deactivatePlugin** [\<>](src/index.js#L11-L11)

Deactivates an active plugin.

_Parameters_

-   `string`: Plugin slug.

<a href="#disableprepublishchecks">#</a> **disablePrePublishChecks** [\<>](src/index.js#L12-L12)

Disables Pre-publish checks.

<a href="#enablepagedialogaccept">#</a> **enablePageDialogAccept** [\<>](src/index.js#L13-L13)

Enables even listener which accepts a page dialog which
may appear when navigating away from Gutenberg.

<a href="#enableprepublishchecks">#</a> **enablePrePublishChecks** [\<>](src/index.js#L14-L14)

Enables Pre-publish checks.

<a href="#ensuresidebaropened">#</a> **ensureSidebarOpened** [\<>](src/index.js#L15-L15)

Verifies that the edit post sidebar is opened, and if it is not, opens it.

_Returns_

-   `Promise`: Promise resolving once the edit post sidebar is opened.

<a href="#findsidebarpaneltogglebuttonwithtitle">#</a> **findSidebarPanelToggleButtonWithTitle** [\<>](src/index.js#L16-L16)

Finds a sidebar panel with the provided title.

_Parameters_

-   `string`: The name of sidebar panel.

_Returns_

-   `?ElementHandle`: Object that represents an in-page DOM element.

<a href="#findsidebarpanelwithtitle">#</a> **findSidebarPanelWithTitle** [\<>](src/index.js#L17-L17)

Finds the button responsible for toggling the sidebar panel with the provided title.

_Parameters_

-   `string`: The name of sidebar panel.

_Returns_

-   `?ElementHandle`: Object that represents an in-page DOM element.

<a href="#getallblockinserteritemtitles">#</a> **getAllBlockInserterItemTitles** [\<>](src/index.js#L18-L18)

Returns an array of strings with all inserter item titles.

_Returns_

-   `Promise`: Promise resolving with an array containing all inserter item titles.

<a href="#getallblocks">#</a> **getAllBlocks** [\<>](src/index.js#L19-L19)

Returns an array with all blocks; Equivalent to calling wp.data.select( 'core/editor' ).getBlocks();

_Returns_

-   `Promise`: Promise resolving with an array containing all blocks in the document.

<a href="#getavailableblocktransforms">#</a> **getAvailableBlockTransforms** [\<>](src/index.js#L20-L20)

Returns an array of strings with all block titles,
that the current selected block can be transformed into.

_Returns_

-   `Promise`: Promise resolving with an array containing all possible block transforms

<a href="#getblocksetting">#</a> **getBlockSetting** [\<>](src/index.js#L21-L21)

Returns a string containing the block title associated with the provided block name.

_Parameters_

-   `string`: Block name.
-   `string`: Block setting e.g: title, attributes....

_Returns_

-   `Promise`: Promise resolving with a string containing the block title.

<a href="#geteditedpostcontent">#</a> **getEditedPostContent** [\<>](src/index.js#L22-L22)

Returns a promise which resolves with the edited post content (HTML string).

_Returns_

-   `Promise`: Promise resolving with post content markup.

<a href="#hasblockswitcher">#</a> **hasBlockSwitcher** [\<>](src/index.js#L23-L23)

Returns a boolean indicating if the current selected block has a block switcher or not.

_Returns_

-   `Promise`: Promise resolving with a boolean.

<a href="#insertblock">#</a> **insertBlock** [\<>](src/index.js#L24-L24)

Opens the inserter, searches for the given term, then selects the first
result that appears.

_Parameters_

-   `string`: The text to search the inserter for.
-   `string`: The inserter panel to open (if it's closed by default).

<a href="#installplugin">#</a> **installPlugin** [\<>](src/index.js#L25-L25)

Installs a plugin from the WP.org repository.

_Parameters_

-   `string`: Plugin slug.
-   `?string`: If the plugin is not findable by its slug use an alternative term to search.

<a href="#iscurrenturl">#</a> **isCurrentURL** [\<>](src/index.js#L26-L26)

Checks if current URL is a WordPress path.

_Parameters_

-   `string`: String to be serialized as pathname.
-   `?string`: String to be serialized as query portion of URL.

_Returns_

-   `boolean`: Boolean represents whether current URL is or not a WordPress path.

<a href="#isindefaultblock">#</a> **isInDefaultBlock** [\<>](src/index.js#L27-L27)

Checks if the block that is focused is the default block.

_Returns_

-   `Promise`: Promise resolving with a boolean indicating if the focused block is the default block.

<a href="#loginuser">#</a> **loginUser** [\<>](src/index.js#L28-L28)

Performs log in with specified username and password.

_Parameters_

-   `?string`: String to be used as user credential.
-   `?string`: String to be used as user credential.

<a href="#mockortransform">#</a> **mockOrTransform** [\<>](src/index.js#L52-L52)

Mocks a request with the supplied mock object, or allows it to run with an optional transform, based on the
deserialised JSON response for the request.

_Parameters_

-   `function`: function that returns true if the request should be mocked.
-   `Object`: A mock object to wrap in a JSON response, if the request should be mocked.
-   `(function|undefined)`: An optional function that transforms the response's object before the response is used.

_Returns_

-   `Promise`: Promise that uses `mockCheck` to see if a request should be mocked with `mock`, and optionally transforms the response with `responseObjectTransform`.

<a href="#observefocusloss">#</a> **observeFocusLoss** [\<>](src/index.js#L29-L29)

Binds to the document on page load which throws an error if a `focusout`
event occurs without a related target (i.e. focus loss).

<a href="#openallblockinsertercategories">#</a> **openAllBlockInserterCategories** [\<>](src/index.js#L30-L30)

Opens all block inserter categories.

<a href="#opendocumentsettingssidebar">#</a> **openDocumentSettingsSidebar** [\<>](src/index.js#L31-L31)

Clicks on the button in the header which opens Document Settings sidebar when it is closed.

<a href="#openglobalblockinserter">#</a> **openGlobalBlockInserter** [\<>](src/index.js#L32-L32)

Opens the global block inserter.

<a href="#openpublishpanel">#</a> **openPublishPanel** [\<>](src/index.js#L33-L33)

Opens the publish panel.

<a href="#presskeytimes">#</a> **pressKeyTimes** [\<>](src/index.js#L34-L34)

Presses the given keyboard key a number of times in sequence.

_Parameters_

-   `string`: Key to press.
-   `number`: Number of times to press.

_Returns_

-   `Promise`: Promise resolving when key presses complete.

<a href="#presskeywithmodifier">#</a> **pressKeyWithModifier** [\<>](src/index.js#L35-L35)

Performs a key press with modifier (Shift, Control, Meta, Alt), where each modifier
is normalized to platform-specific modifier.

_Parameters_

-   `string`: Modifier key.
-   `string`: Key to press while modifier held.

<a href="#publishpost">#</a> **publishPost** [\<>](src/index.js#L36-L36)

Publishes the post, resolving once the request is complete (once a notice
is displayed).

_Returns_

-   `Promise`: Promise resolving when publish is complete.

<a href="#publishpostwithprepublishchecksdisabled">#</a> **publishPostWithPrePublishChecksDisabled** [\<>](src/index.js#L37-L37)

Publishes the post without the pre-publish checks,
resolving once the request is complete (once a notice is displayed).

_Returns_

-   `Promise`: Promise resolving when publish is complete.

<a href="#savedraft">#</a> **saveDraft** [\<>](src/index.js#L38-L38)

Saves the post as a draft, resolving once the request is complete (once the
"Saved" indicator is displayed).

_Returns_

-   `Promise`: Promise resolving when draft save is complete.

<a href="#searchforblock">#</a> **searchForBlock** [\<>](src/index.js#L39-L39)

Search for block in the global inserter

_Parameters_

-   `string`: The text to search the inserter for.

<a href="#selectblockbyclientid">#</a> **selectBlockByClientId** [\<>](src/index.js#L40-L40)

Given the clientId of a block, selects the block on the editor.

_Parameters_

-   `string`: Identified of the block.

<a href="#setbrowserviewport">#</a> **setBrowserViewport** [\<>](src/index.js#L41-L41)

Sets browser viewport to specified type.

_Parameters_

-   `string`: String to represent dimensions type; can be either small or large.

<a href="#setpostcontent">#</a> **setPostContent** [\<>](src/index.js#L42-L42)

Sets code editor content

_Parameters_

-   `string`: New code editor content.

_Returns_

-   `Promise`: Promise resolving with an array containing all blocks in the document.

<a href="#setupresponsemocking">#</a> **setUpResponseMocking** [\<>](src/index.js#L52-L52)

Sets up mock checks and responses. Accepts a list of mock settings with the following properties:

-   `match`: function to check if a request should be mocked.
-   `onRequestMatch`: async function to respond to the request.

_Usage_

```js
const MOCK_RESPONSES = [
  {
    match: isEmbedding( 'https://wordpress.org/gutenberg/handbook/' ),
    onRequestMatch: JSONResponse( MOCK_BAD_WORDPRESS_RESPONSE ),
  },
  {
    match: isEmbedding( 'https://wordpress.org/gutenberg/handbook/block-api/attributes/' ),
    onRequestMatch: JSONResponse( MOCK_EMBED_WORDPRESS_SUCCESS_RESPONSE ),
  }
];
setUpResponseMocking( MOCK_RESPONSES );
```

If none of the mock settings match the request, the request is allowed to continue.

_Parameters_

-   `Array`: Array of mock settings.

<a href="#switcheditormodeto">#</a> **switchEditorModeTo** [\<>](src/index.js#L43-L43)

Switches editor mode.

_Parameters_

-   `string`: String editor mode.

<a href="#switchusertoadmin">#</a> **switchUserToAdmin** [\<>](src/index.js#L44-L44)

Switches the current user to the admin user (if the user
running the test is not already the admin user).

<a href="#switchusertotest">#</a> **switchUserToTest** [\<>](src/index.js#L45-L45)

Switches the current user to whichever user we should be
running the tests as (if we're not already that user).

<a href="#togglescreenoption">#</a> **toggleScreenOption** [\<>](src/index.js#L46-L46)

Toggles the screen option with the given label.

_Parameters_

-   `string`: The label of the screen option, e.g. 'Show Tips'.
-   `?boolean`: If true, turns the option on. If false, off. If undefined, the option will be toggled.

<a href="#transformblockto">#</a> **transformBlockTo** [\<>](src/index.js#L47-L47)

Converts editor's block type.

_Parameters_

-   `string`: Block name.

<a href="#uninstallplugin">#</a> **uninstallPlugin** [\<>](src/index.js#L48-L48)

Uninstalls a plugin.

_Parameters_

-   `string`: Plugin slug.

<a href="#visitadminpage">#</a> **visitAdminPage** [\<>](src/index.js#L49-L49)

Visits admin page; if user is not logged in then it logging in it first, then visits admin page.

_Parameters_

-   `string`: String to be serialized as pathname.
-   `string`: String to be serialized as query portion of URL.

<a href="#waitforwindowdimensions">#</a> **waitForWindowDimensions** [\<>](src/index.js#L50-L50)

Function that waits until the page viewport has the required dimensions.
It is being used to address a problem where after using setViewport the execution may continue,
without the new dimensions being applied.
<https://github.com/GoogleChrome/puppeteer/issues/1751>

_Parameters_

-   `number`: Width of the window.
-   `height`: Height of the window.


<!-- END TOKEN(Autogenerated API docs) -->

<br/><br/><p align="center"><img src="https://s.w.org/style/images/codeispoetry.png?1" alt="Code is Poetry." /></p>
