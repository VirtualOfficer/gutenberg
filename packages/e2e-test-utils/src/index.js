export { activatePlugin } from './activate-plugin';
export { arePrePublishChecksEnabled } from './are-pre-publish-checks-enabled';
export { clearLocalStorage } from './clear-local-storage';
export { clickBlockAppender } from './click-block-appender';
export { clickBlockToolbarButton } from './click-block-toolbar-button';
export { clickButton } from './click-button';
export { clickOnCloseModalButton } from './click-on-close-modal-button';
export { clickOnMoreMenuItem } from './click-on-more-menu-item';
export { createNewPost } from './create-new-post';
export { createURL } from './create-url';
export { deactivatePlugin } from './deactivate-plugin';
export { disablePrePublishChecks } from './disable-pre-publish-checks';
export { dragAndResize } from './drag-and-resize';
export { enablePageDialogAccept } from './enable-page-dialog-accept';
export { enablePrePublishChecks } from './enable-pre-publish-checks';
export { ensureSidebarOpened } from './ensure-sidebar-opened';
export { externalWrapperHasFocus } from './external-wrapper-has-focus';
export { findSidebarPanelToggleButtonWithTitle } from './find-sidebar-panel-toggle-button-with-title';
export { findSidebarPanelWithTitle } from './find-sidebar-panel-with-title';
export { getAllBlockInserterItemTitles } from './get-all-block-inserter-item-titles';
export { getAllBlocks } from './get-all-blocks';
export { getAvailableBlockTransforms } from './get-available-block-transforms';
export { getBlockSetting } from './get-block-setting';
export { getEditedPostContent } from './get-edited-post-content';
export { getElementSelectorList } from './get-element-list';
export { hasBlockSwitcher } from './has-block-switcher';
export { insertAndPopulateBlock } from './insert-and-populate-block';
export { insertBlock } from './insert-block';
export { inserterToggleHasFocus } from './inserter-toggle-has-focus';
export { installPlugin } from './install-plugin';
export { isCurrentURL } from './is-current-url';
export { isInDefaultBlock } from './is-in-default-block';
export { loginUser } from './login-user';
export { navigateToContentEditorTop } from './navigate-to-content-editor-top';
export { observeFocusLoss } from './observe-focus-loss';
export { openAllBlockInserterCategories } from './open-all-block-inserter-categories';
export { openDocumentSettingsSidebar } from './open-document-settings-sidebar';
export { openGlobalBlockInserter } from './open-global-block-inserter';
export { openPublishPanel } from './open-publish-panel';
export { pressKeyTimes } from './press-key-times';
export { pressKeyWithModifier } from './press-key-with-modifier';
export { publishPost } from './publish-post';
export { publishPostWithPrePublishChecksDisabled } from './publish-post-with-pre-publish-checks-disabled';
export { saveDraft } from './save-draft';
export { searchForBlock } from './search-for-block';
export { selectBlockByClientId } from './select-block-by-client-id';
export { setBrowserViewport } from './set-browser-viewport';
export { setPostContent } from './set-post-content';
export { switchEditorModeTo } from './switch-editor-mode-to';
export { switchUserToAdmin } from './switch-user-to-admin';
export { switchUserToTest } from './switch-user-to-test';
export { tabThroughBlock } from './tab-through-block';
export { tabThroughBlockMoverControl } from './tab-through-block-mover-control';
export { tabThroughBlockToolbar } from './tab-through-block-toolbar';
export { tabThroughFileBlock } from './tab-through-file-block';
export { tabThroughPlaceholderButtons } from './tab-through-placeholder-buttons';
export { tabThroughTextBlock } from './tab-through-text-block';
export { textContentAreas } from './text-content-areas';
export { textContentAreasHaveFocus } from './text-content-areas-have-focus';
export { toggleScreenOption } from './toggle-screen-option';
export { transformBlockTo } from './transform-block-to';
export { uninstallPlugin } from './uninstall-plugin';
export { visitAdminPage } from './visit-admin-page';
export { waitForWindowDimensions } from './wait-for-window-dimensions';

export * from './mocks';
