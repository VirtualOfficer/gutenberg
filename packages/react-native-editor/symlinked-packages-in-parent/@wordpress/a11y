../../../packages/a11y/build/