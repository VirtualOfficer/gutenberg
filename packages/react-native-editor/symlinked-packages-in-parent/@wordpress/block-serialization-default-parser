../../../packages/block-serialization-default-parser/build/