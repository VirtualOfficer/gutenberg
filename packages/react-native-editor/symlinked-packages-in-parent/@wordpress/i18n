../../../packages/i18n/build/