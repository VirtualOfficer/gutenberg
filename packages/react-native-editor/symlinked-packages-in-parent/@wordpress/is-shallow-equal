../../../packages/is-shallow-equal/build/