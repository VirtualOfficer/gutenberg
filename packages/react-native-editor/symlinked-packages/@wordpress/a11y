../../gutenberg/packages/a11y/src/