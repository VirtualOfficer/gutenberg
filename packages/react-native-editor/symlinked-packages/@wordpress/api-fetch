../../gutenberg/packages/api-fetch/src/