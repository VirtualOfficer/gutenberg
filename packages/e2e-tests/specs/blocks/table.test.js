/**
 * WordPress dependencies
 */
import { createNewPost, insertBlock, getEditedPostContent } from '@wordpress/e2e-test-utils';

describe( 'Table', () => {
	beforeEach( async () => {
		await createNewPost();
	} );

	it( 'displays a form for choosing the row and column count of the table', async () => {
		await insertBlock( 'Table' );

		// Check for existence of the column count field.
		const columnCountLabel = await page.$x( "//div[@data-type='core/table']//label[text()='Column Count']" );
		expect( columnCountLabel ).toHaveLength( 1 );

		// Modify the column count.
		await columnCountLabel[ 0 ].click();
		const currentColumnCount = await page.evaluate( () => document.activeElement.value );
		expect( currentColumnCount ).toBe( '2' );
		await page.keyboard.press( 'Backspace' );
		await page.keyboard.type( '5' );

		// // Check for existence of the row count field.
		const rowCountLabel = await page.$x( "//div[@data-type='core/table']//label[text()='Row Count']" );
		expect( rowCountLabel ).toHaveLength( 1 );

		// // Modify the row count.
		await rowCountLabel[ 0 ].click();
		const currentRowCount = await page.evaluate( () => document.activeElement.value );
		expect( currentRowCount ).toBe( '2' );
		await page.keyboard.press( 'Backspace' );
		await page.keyboard.type( '10' );

		// // Create the table.
		const createButton = await page.$x( "//div[@data-type='core/table']//button[text()='Create']" );
		await createButton[ 0 ].click();

		// // Expect the post content to have a correctly sized table.
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'allows text to by typed into cells', async () => {
		await insertBlock( 'Table' );

		// Create the table.
		const createButton = await page.$x( "//div[@data-type='core/table']//button[text()='Create']" );
		await createButton[ 0 ].click();

		// Click the first cell and add some text.
		await page.click( '.wp-block-table__cell-content' );
		await page.keyboard.type( 'This' );

		// Tab to the next cell and add some text.
		await page.keyboard.press( 'Tab' );
		await page.keyboard.type( 'is' );

		// Tab to the next cell and add some text.
		await page.keyboard.press( 'Tab' );
		await page.keyboard.type( 'table' );

		// Tab to the next cell and add some text.
		await page.keyboard.press( 'Tab' );
		await page.keyboard.type( 'block' );

		// Expect the post content to have a correctly sized table.
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );
} );
