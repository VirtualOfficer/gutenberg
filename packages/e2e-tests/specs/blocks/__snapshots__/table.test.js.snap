// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table allows text to by typed into cells 1`] = `
"<!-- wp:table -->
<table class=\\"wp-block-table\\"><tbody><tr><td>This</td><td>is</td></tr><tr><td>table</td><td>block</td></tr></tbody></table>
<!-- /wp:table -->"
`;

exports[`Table displays a form for choosing the row and column count of the table 1`] = `
"<!-- wp:table -->
<table class=\\"wp-block-table\\"><tbody><tr><td></td><td></td><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td><td></td><td></td></tr></tbody></table>
<!-- /wp:table -->"
`;
